### DISCUSSION TITLE
Migrating to Parcel2: Issues with HMR, Babel, and TS

### DISCUSSION THREAD

**Discussion by sezgi** [parcel-bundler/parcel#7006] [Q&A]
Running into a few issues trying to migrate a React/TS app from Parcel1 to Parcel2.

### HMR not working
```
WebSocket connection to 'ws://localhost:8888/' failed:
(error message is `undefined`)
[parcel] 🚨 Connection to the HMR server was lost
```

### Can't remove Babel
I've managed to remove most of Babel (kept `browserslist` in `package.json`), but when I remove `@babel/preset-react`, I get this error:
```
Build failed.

@parcel/transformer-js: Unexpected token `)`. Expected this, import, async, function, [ for array literal, { for object literal, @ for decorator, function, class, null, true, false, number, bigint, string, regexp, ` for template literal, (, or an identifier

  1 |   const example = useMemo(
> 2 |     () =>
>   |         ^
```

### Can't remove `tsc`
When I remove `@parcel/transformer-typescript-tsc` from `.parcelrc`, I get this on page load:
`Uncaught ReferenceError: Cannot access 'EXAMPLE_CONST' before initialization`

**Comment by sezgi** [comment]
This might be the same HMR issue:
https://github.com/parcel-bundler/parcel/issues/6994

Using a different `--hmr-port` seems to work. Doing that for now.

**Comment by devongovett** [comment]
Can you provide a code sample that reproduces your issue please?
